src/contracts/services/lend.ts

```diff
@@ -0,0 +1,512 @@
+import {
+  BatchQueryParams,
+  Contract,
+  BatchQueryParsedResponse,
+  MinBlockHeightValidationOptions,
+} from '~/types';
+import { batchQuery$ } from '~/contracts/services/batchQuery';
+import {
+  map,
+  first,
+  switchMap,
+  lastValueFrom,
+} from 'rxjs';
+import {
+  VaultResponse,
+  VaultsResponse,
+  NormalizationFactor,
+  PositionResponse,
+  PositionsResponse,
+} from '~/types/contracts/lend/response';
+import {
+  convertCoinFromUDenom,
+  getActiveQueryClient$,
+} from '~/index';
+import {
+  Vaults,
+  Vault,
+  BatchVaults,
+  VaultUserData,
+  VaultsUserData,
+  BatchVaultsUserData,
+  VaultVersion,
+  LendVaultRegistryContract,
+} from '~/types/contracts/lend/model';
+import {
+  msgGetVault,
+  msgGetVaults,
+  msgGetVaultUserPosition,
+  msgGetVaultUserPositions,
+} from '~/contracts/definitions/lend';
+import { sendSecretClientContractQuery$ } from '~/client/services/clientServices';
+import { AccountPermit } from '~/types/permit';
+import BigNumber from 'bignumber.js';
+
+/**
+* Parse lend vault response
+*/
+function parseLendVault(vault: VaultResponse, vaultVersion: VaultVersion) {
```

Comment:
Can we make vaultVersion optional and have it default to v3, the most up to date version?

Reply:
Is there really no way to programatically determine the version?

Reply:
there is no way to programatically determine, unfortunately.... that would have been a great feature that should have been included when we started making new vaults.

Reply:
I'd rather not set defaults.